New app mod flow doc (#48543)
dotnet/docs: docs/azure/migration/appmod/quickstart.md

description: Get started with GitHub Copilot app modernization for .NET
ms.topic: concept-article
ms.custom: devx-track-dotnet
ms.date: 9/17/2025
author: alexwolfmsft
ms.author: alexwolf
---

# Quickstart: assess and migrate a .NET Project to Azure

In this quickstart, you assess and migrate a .NET project using [GitHub Copilot app modernization for .NET (Preview)](https://aka.ms/ghcp-appmod/dotNET). You will complete the following tasks:

- Install and configure the GitHub Copilot app modernization for .NET extension
- Assess a sample project (Contoso University)
- Start the migration process

## Prerequisites

> Before you begin, ensure follow the prerequisites of [https://aka.ms/ghcp-appmod/dotNET#prerequisites].

Reviewer: ```suggestion
Before you begin, [complete the preqrequisites](https://aka.ms/ghcp-appmod/dotNET#prerequisites).
```

5. A new Copilot chat session opens and displays the welcome message. The assessment automatically begins analyzing your project for migration readiness.

    :::image type="content" source="media/AssessmentInProcess.png" alt-text="A screenshot showing assessment in progress with status indicators.":::

6. When the assessment completes, you can view a comprehensive assessment report UI page and receive a list of migration tasks in the chat window.

    :::image type="content" source="media/assessment-report.png" alt-text="A screenshot showing the generated assessment report with detailed findings.":::

## Start migration

GitHub Copilot app modernization for .NET (Preview) includes [predefined tasks](predefined-tasks.md) for common migration scenarios, following Microsoft's best practices.

### Start a migration task

You can start a migration task in one of the following ways:

**Option 1. Run from the Assessment Report**

Select the **Run Task** button in the Assessment Report generated in the previous step to start a migration task.

**Option 2. Send in Copilot Chat**

Send the migration task number (e.g., 1.1) or its name in the chat.

>     :::image type="content" source="media/quickstart_chat_experience.png" alt-text="A screenshot showing sending the message in the Copilot Chat to start a migration task.":::

Reviewer: ```suggestion

:::image type="content" source="media/quickstart_chat_experience.png" alt-text="A screenshot showing sending the message in the Copilot Chat to start a migration task.":::
```

### Plan and progress tracker generation

> - When you start the migration, GitHub Copilot begins a session named "App modernization: migrate from <source technology> to <target technology>" in agent mode with predefined prompts.

Reviewer: ```suggestion
- When you start the migration, GitHub Copilot begins a session named "App modernization: migrate from `<source technology>` to `<target technology>`" in agent mode with predefined prompts.
```

  - Apply configuration changes
  - Make code changes
  - Build the solution, fix all compilation and configuration errors, and ensure a successful build
  - Security vulnerability fix

## Default chat messages

> The GitHub Copilot app modernization for .NET extension provides default chat message options to streamline your workflow.
>     :::image type="content" source="media/quickstart_followup.png" alt-text="A screenshot showing default chat message options in the Copilot Chat.":::

Reviewer: ```suggestion
The GitHub Copilot app modernization for .NET extension provides default chat message options to streamline your workflow.

:::image type="content" source="media/quickstart_followup.png" alt-text="A screenshot showing default chat message options in the Copilot Chat.":::
```
